api/src/services/opportunities_v0_1/search_opportunities.py

```diff
@@ -113,18 +104,49 @@ def _add_filters(
     return stmt
 
 
+def _add_order_by(
+    stmt: Select[tuple[Opportunity]], pagination: PaginationParams
+) -> Select[tuple[Opportunity]]:
+    # This generates an order by command like:
+    #
+    #   ORDER BY opportunity.opportunity_id DESC NULLS LAST
+
+    # This determines whether we use ascending or descending when building the query
+    sort_fn = asc if pagination.is_ascending else desc
+
+    match pagination.order_by:
+        case "opportunity_id":
+            field: InstrumentedAttribute = Opportunity.opportunity_id
+        case "opportunity_number":
+            field = Opportunity.opportunity_number
+        case "opportunity_title":
+            field = Opportunity.opportunity_title
+        case "post_date":
+            field = OpportunitySummary.post_date
+        case "close_date":
+            field = OpportunitySummary.close_date
+        case "agency_code":
+            field = Opportunity.agency
+        case _:
+            # If this exception happens, it means our API schema
+            # allows for values we don't have implemented. This
+            # means we can't determine how to sort / need to correct
+            # the mismatch.
+            msg = f"Unconfigured sort_by parameter {pagination.order_by} provided, cannot determine how to sort."
+            raise Exception(msg)
```

Comment:
This seems like is should raise a HTTP 400, not a HTTP 500? Do you have a class for that?

Reply:
I wanted to return a 500 here, if this happens, it is because of a bug in our code. This means our API allows you to pass in a field to sort by that we haven't implemented.